caching/caching.go

```diff
@@ -90,6 +94,46 @@ func NewCachingMiddleware(mw *cachingMiddleware) middleware.InitializeMiddleware
 	})
 }
 
+func (d *cachingMiddleware) startAsyncBatchWriter() {
+	const maxBufferSize = 10000 // TODO think about this number. Should we make configurable or larger/smaller?
```

Comment:
I think this number is pretty massive overkill. I like the idea of eventually making it configurable but for now it looks like a smaller number would suffice. Maybe start with 100?